openpype/hosts/maya/plugins/publish/collect_animation.py

```diff
@@ -59,16 +59,28 @@ def process(self, instance):
             instance.data["families"].append("publish.farm")
 
         # Collect user defined attributes.
-        if not instance.data.get("includeUserDefinedAttributes", False):
-            return
-
-        user_defined_attributes = set()
-        for node in hierarchy:
-            attrs = cmds.listAttr(node, userDefined=True) or list()
-            shapes = cmds.listRelatives(node, shapes=True) or list()
-            for shape in shapes:
-                attrs.extend(cmds.listAttr(shape, userDefined=True) or list())
-
-            user_defined_attributes.update(attrs)
-
-        instance.data["userDefinedAttributes"] = list(user_defined_attributes)
+        if instance.data.get("includeUserDefinedAttributes", False):
+            user_defined_attributes = set()
+            for node in hierarchy:
+                attrs = cmds.listAttr(node, userDefined=True) or list()
+                shapes = cmds.listRelatives(node, shapes=True) or list()
+                for shape in shapes:
+                    attrs.extend(
+                        cmds.listAttr(shape, userDefined=True) or list()
+                    )
+
+                user_defined_attributes.update(attrs)
+
+            instance.data["userDefinedAttributes"] = list(
+                user_defined_attributes
+            )
+
+        # Backwards compatibility for attributes.
+        backwards_mapping = {
+            "write_color_sets": "writeColorSets",
+            "write_face_sets": "writeFaceSets",
+            "include_parent_hierarchy": "includeParentHierarchy",
+            "include_user_defined_attributes": "includeUserDefinedAttributes"
+        }
+        for key, value in backwards_mapping.items():
```

Comment:
Wasn't there a validator which would "destroy" attributes that didn't exist anymore? or would it only create new ones?

Reply:
Only creates new ones, so we can guarantee the new ones are there but cant know whether the old ones are there or not.